templates/shortcode-goal.php

```diff
@@ -3,7 +3,6 @@
  * This template is used to display the goal with [give_goal]
  */
 
-$form_id          = get_the_ID(); // Form ID.
 $form        = new Give_Donate_Form( $form_id );
```

Comment:
I know that the **$form_id** value is passed through the **give_show_goal_progress()** from the **includes/forms/template.php** file, but at first, the look seems stranger to using a variable without initializing that. So, I would suggest initializing it using something like this: `$form_id = isset($form_id) ? $form_id : 0;`

Reply:
I wouldn't advise this. If the `$form_id` comes from a parent context it just needs to be documented. If we check and set the value to 0 then we're just setting ourselves up for a potentially unexpected outcome. I would first document it via:
```php
/**
 * @var int $form_id form id passed from the give_show_goal_progress() context
 */
```

Then, if we feel there's a chance this could be misused, I would do something like:
```php
if ( empty($form_id) ) {
    Log::error(...);
    return false;
}
```
I really don't like returning false, as that rarely means what we think it means, but since it's what's already happening that's fine.